src/main/kotlin/com/openlattice/linking/BackgroundLinkingService.kt

```diff
@@ -59,100 +56,47 @@ class BackgroundLinkingService
-                                                Instant.now().toEpochMilli()
-                                        )
-                                        if (expiration != null && Instant.now().toEpochMilli() >= expiration) {
-                                            logger.info("Refreshing expiration for {}", it)
-                                            //Assume original lock holder died, probably somewhat unsafe
-                                            refreshExpiration(it)
-                                            true
-                                        } else expiration == null
-                                    }
-                            if (forLinking.isNotEmpty()) {
-                                logger.info("Entities needing linking: {}", forLinking)
-                            }
-                            forLinking.asSequence()
-                        }
-                        .chunked(configuration.loadSize)
-                        .forEach { keys ->
-                            candidates.addAll(keys)
-                            logger.info(
-                                    "Queued entities needing linking {} from ({}).",
-                                    keys,
-                                    entitySets.getAll(keys.map { it.entitySetId }.toSet()).values.map { it.name }
-                            )
-                        }
-            }
-        } catch (ex: Exception) {
-            logger.info("Encountered error while updating candidates for linking.", ex)
-        }
+    override fun enqueuerEnabledCheck(): Boolean {
+        return configuration.backgroundLinkingEnabled
     }
-    private val candidates = HazelcastQueue.LINKING_CANDIDATES.getQueue( hazelcastInstance )
-    private val limiter = Semaphore(configuration.parallelism)
 
-    @Suppress("UNUSED")
-    private val linkingWorker = if (isLinkingEnabled()) executor.submit {
-        while (true) {
-            try {
-                generateSequence(candidates::take)
-                        .map { candidate ->
-                            limiter.acquire()
-                            executor.submit {
-                                try {
-                                    logger.info("Linking {}", candidate)
-                                    link(candidate)
-                                } catch (ex: Exception) {
-                                    logger.error("Unable to link $candidate. ", ex)
-                                } finally {
-                                    unlock(candidate)
-                                    limiter.release()
-                                }
-                            }
-                        }.forEach { it.get() }
-            } catch (ex: Exception) {
-                logger.info("Encountered error while linking candidates.", ex)
-            }
-        }
+    override fun workerEnabledCheck(): Boolean {
```

Comment:
This and `enqueuerEnabledCheck()` do the same thing, why have two separate methods?